guides/hack/11-built-in-types/35-enum-class.md

```diff
@@ -85,30 +101,68 @@ enum class EC : int {
 hhvm.hack.lang.enable_enum_classes=1
 ```
 
-If we now look at `EC::A`, its type is `HH\MemberOf<EC, int>`.
+The built-in enum type of `E::A` is just `E`. All we know is that constant `A` is declared within the enum: we know nothing of its underlying type.
 
+But if we look at the enum class `EC::A` its type is `HH\MemberOf<EC, int>`. We know that it's declared within the enum class `EC`, with type `int`.
 
-Let's have a look back at `Names::World`, its type is `HH\MemberOf<Names, HasName>`, and `Names::Bar` has type `HH\MemberOf<Names, ConstName>`.
+## Extending an Existing enum class (Inheritance)
 
-The addition of this type layer is here to give more information about the enumeration, namely its exact type
-(`HasName` vs `IHasName`) and from which enum class it comes from (`Names`).
-`HH\MemberOf` is designed to allow a transparent access to the underlying value: `HH\MemberOf<Names, HasName> <: HasName`.
-This means that this layer can be ignored if one doesn’t need the additional information, and `Names::Hello` can be used directly as an instance of the `HasName` class:
+Enum classes can be composed together, as long as they implement the same base type:
```

Comment:
composition is a distinct concept to inheritance, and we should avoid any risk of confusing the two; or perhaps say "because the values are included, this is sometimes referred to as 'enum composition'"

Reply:
Indeed, let's stick to inheritance (as I've been asked to work on another mean of composition :D)